mdpow/analysis/ensemble.py

```diff
@@ -0,0 +1,499 @@
+
+        Uses the same
+        `selection commands <https://docs.mdanalysis.org/stable/documentation_pages/selections.html>`_
+        as MDAnalysis."""
+        selections = {}
+        for key in self.keys():
+            try:
+                ag = self[key].select_atoms(*args, **kwargs)
+            except SelectionError as err:
+                logger.error("%r on system %r with selection settings %r %r", err, key, args, kwargs)
+                raise
+            else:
+                selections[key] = ag
+        return EnsembleAtomGroup(selections, ensemble_dir=self._ensemble_dir)
+
+    def select_systems(self, keys=None, solvents=None, interactions=None,
+                       lambdas=None, lambda_range=None):
+        """Select specific subset of systems and returns them in an Ensemble.
+
+        This can be accomplished in two ways, by specific keys, or by
+        specifying the desired system attributes solvents, interactions and
+        lambdas. All arguments are stored in list form.
+
+        Specific key workflow example::
+
+            Ens = Ensemble(dirname='Mol')
+            w_v_0_25 = Ens.select_systems(keys=[('water', 'VDW', '0000'),
+                                                ('water', 'VDW', '0025')]
+
+        For the system attributes workflow there are two ways of selecting lambdas,
+        the :param lambdas: keyword saves specific lambdas or the :param lambda_range:
+        which saves the lambdas that fall within the given range.
+
+        Specific lambdas example::
+
+            Ens = Ensemble(dirname='Mol')
+            w_v_0_25 = Ens.select_systems(solvents=['water'], interactions=['VDW'],
+                                          lambdas=['0000', '0025'])
+        Range of lambdas example::
+
+            Ens = Ensemble(dirname='Mol')
+            w_v = Ens.select_systems(solvents=['water'], interactions=['VDW'],
+                                     lambda_range=[0, 1])
+
+        """
+        new_ens = Ensemble()
+        new_key = []
+        if not keys is None:
+            # Selection by giving keys
+            new_key = keys
+        elif not solvents is None:
+            # Selection by attributes
+            for s in solvents:
+                if not interactions is None:
+                    for i in interactions:
+                        if not lambdas is None:
+                            # Selecting specific lambdas
+                            for l in lambdas:
+                                new_key.append((s, i, l))
+                        elif not lambda_range is None:
+                            # Selecting range of lambdas
+                            for k in self.keys():
+                                if lambda_range[0] <= int(k[2])/1000 <= lambda_range[1]:
+                                    new_key.append((s, i, k[2]))
+        for k in new_key:
+            logger.info('adding system %r to ensemble', k)
+            new_ens.add_system(k, universe=self[k])
```

Comment:
nice, that's how I envisage add_system() being used

Reply:
thanks, I'm glad Ensemble has been reasonable modular.